src/core/server/config/deprecation/core_deprecations.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { has } from 'lodash';
+import { ConfigDeprecationProvider, ConfigDeprecation } from './types';
+
+const configPathDeprecation: ConfigDeprecation = (settings, fromPath, log) => {
+  if (has(process.env, 'CONFIG_PATH')) {
+    log(
+      `Environment variable CONFIG_PATH is deprecated. It has been replaced with KIBANA_PATH_CONF pointing to a config folder`
+    );
+  }
+  return settings;
+};
+
+const dataPathDeprecation: ConfigDeprecation = (settings, fromPath, log) => {
+  if (has(process.env, 'DATA_PATH')) {
+    log(
+      `Environment variable "DATA_PATH" will be removed.  It has been replaced with kibana.yml setting "path.data"`
```

Comment:
shouldn't we remove core deprecations from the legacy now?

Reply:
Good question, and I'm not sure of the answer to be honest.

There is one deprecation I did not migrate yet from `src/legacy/server/config/transform_deprecations.js`, the one regarding CSP rules. in legacy, the csp config is located at `csp`. In https://github.com/elastic/kibana/pull/52161, It seems we moved it to `server.csp`, so I was not sure what to do about it.

But I could at least remove all the others from legacy I think

Reply:
Actually I migrated the `csp` deprecations. Btw the logic inside it looks like more like actual validation that deprecation, we might want to move it to config validation at some point.

That was the last thing blocking, I should be able to suppress `src/legacy/server/config/transform_deprecations.js` and it's usages now.

Reply:
https://github.com/elastic/kibana/pull/52251/commits/b9e0c6ca6dc3d72edbe6eea5302c2a5e746f8fe2 removes the legacy core deprecations.